pinot-core/src/main/java/org/apache/pinot/core/segment/index/readers/text/LuceneTextIndexReader.java

```diff
@@ -148,8 +153,12 @@ private MutableRoaringBitmap getPinotDocIds(MutableRoaringBitmap luceneDocIds) {
     try {
       while (luceneDocIDIterator.hasNext()) {
         int luceneDocId = luceneDocIDIterator.next();
-        Document document = _indexSearcher.doc(luceneDocId);
-        int pinotDocId = Integer.valueOf(document.get(LuceneTextIndexCreator.LUCENE_INDEX_DOC_ID_COLUMN_NAME));
+        Integer pinotDocId = _luceneDocIDToPinotDocIDCache.get(luceneDocId);
```

Comment:
Please use `computeIfAbsent` instead

Reply:
I am not using this method anymore. The mapping is built once during segment load and memory mapped.

Reply:
Also added a TODO to consider building mapping on-the-fly during query processing -- we can have two methods and make it configurable based on number of docs per segment.